plugin.video.vrt.nu/resources/lib/kodiwrapper.py

```diff
@@ -437,61 +486,83 @@ def get_path(self, path):
 
     def listdir(self, path):
         ''' Return all files in a directory (using xbmcvfs)'''
-        import xbmcvfs
-        return xbmcvfs.listdir(path)
+        from xbmcvfs import listdir
+        return listdir(path)
 
     def mkdir(self, path):
         ''' Create a directory (using xbmcvfs) '''
-        import xbmcvfs
+        from xbmcvfs import mkdir
         self.log_notice("Create directory '%s'." % path, 'Debug')
-        return xbmcvfs.mkdir(path)
+        return mkdir(path)
 
     def mkdirs(self, path):
         ''' Create directory including parents (using xbmcvfs) '''
-        import xbmcvfs
+        from xbmcvfs import mkdirs
         self.log_notice("Recursively create directory '%s'." % path, 'Debug')
-        return xbmcvfs.mkdirs(path)
+        return mkdirs(path)
 
     def check_if_path_exists(self, path):
         ''' Whether the path exists (using xbmcvfs)'''
-        import xbmcvfs
-        return xbmcvfs.exists(path)
+        from xbmcvfs import exists
+        return exists(path)
 
     @contextmanager
     def open_file(self, path, flags='r'):
         ''' Open a file (using xbmcvfs) '''
-        import xbmcvfs
-        f = xbmcvfs.File(path, flags)
+        from xbmcvfs import File
+        f = File(path, flags)
         yield f
         f.close()
 
     def stat_file(self, path):
         ''' Return information about a file (using xbmcvfs) '''
-        import xbmcvfs
-        return xbmcvfs.Stat(path)
+        from xbmcvfs import Stat
+        return Stat(path)
 
     def delete_file(self, path):
         ''' Remove a file (using xbmcvfs) '''
-        import xbmcvfs
+        from xbmcvfs import delete
         self.log_notice("Delete file '%s'." % path, 'Debug')
-        return xbmcvfs.delete(path)
-
-    def md5(self, path):
-        ''' Return an MD5 checksum of a file '''
+        return delete(path)
+
+    def delete_cached_thumbnail(self, url):
```

Comment:
You are not allowed to change/delete outside your own directories. You should use the proper Kodi API using `Textures.GetTextures` and `Textures.RemoveTexture` (See https://kodi.wiki/view/JSON-RPC_API/v8). 

I understand the reason behind the cache removal, but you are not updating the texture database like this. You could also try to update the texture by adding a random URL parameters? I guess it should be seen as a new thumb?  Not 100% sure.

Reply:
cc @mediaminister

Reply:
Okay, I rewrote `delete_cached_thumbnail` function using Kodi JSON-RPC API to comply with the rules: https://github.com/pietje666/plugin.video.vrt.nu/pull/443
@dagwieers: Can you add this to this release?

Reply:
I think we are going for a v2.0.1 release instead. Since v2.0.0 was already cut from wood.
(And handed out)